### ISSUE TITLE
v1.7.6 can't output to local port

### ISSUE THREAD

**Issue by nokute78** [fluent/fluent-bit#3556]
## Bug Report

**Describe the bug**

v1.7.6 can't send data to local port.
I confirmed `out_loki` and `out_http`. 

**To Reproduce**
- Rubular link if applicable: N/A
- Example log message if applicable:
```
$ bin/fluent-bit -i cpu -o loki
Fluent Bit v1.7.6
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/05/26 17:27:20] [ info] [engine] started (pid=77696)
[2021/05/26 17:27:20] [ info] [storage] version=1.1.1, initializing...
[2021/05/26 17:27:20] [ info] [storage] in-memory
[2021/05/26 17:27:20] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/05/26 17:27:20] [ info] [output:loki:loki.0] configured, hostname=127.0.0.1:3100
[2021/05/26 17:27:20] [ info] [sp] stream processor started
[2021/05/26 17:27:36] [error] [upstream] connection #-1 to 127.0.0.1:3100 timed out after 10 seconds
```
- Steps to reproduce the problem:
Listen port 3100 and send data.

1. `nc -l 3100`
2. `fluent-bit -i cpu -o loki` or `fluent-bit -i cpu -o http -p port=3100`

**Expected behavior**

nc  can get data like this.
```
$ nc -l 3100
POST /loki/api/v1/push HTTP/1.1
Host: 127.0.0.1:3100
Content-Length: 596
User-Agent: Fluent-Bit
Content-Type: application/json

{"streams":[{"stream":{"job":"fluent-bit"},"values":[["1622018212064598380","{\"cpu_p\":2.0,\"user_p\":2.0,\"system_p\":0.0,\"cpu0.p_cpu\":2.0,\"cpu0.p_user\":2.0,\"cpu0.p_system\":0.0}"],["1622018213064627550","{\"cpu_p\":1.0,\"user_p\":0.0,\"system_p\":1.0,\"cpu0.p_cpu\":1.0,\"cpu0.p_user\":0.0,\"cpu0.p_system\":1.0}"],["1622018214064919866","{\"cpu_p\":3.0,\"user_p\":3.0,\"system_p\":0.0,\"cpu0.p_cpu\":3.0,\"cpu0.p_user\":3.0,\"cpu0.p_system\":0.0}"],["1622018215064648479","{\"cpu_p\":3.0,\"user_p\":2.0,\"system_p\":1.0,\"cpu0.p_cpu\":3.0,\"cpu0.p_user\":2.0,\"cpu0.p_system\":1.0}"]]}]}
```

**Screenshots**

N/A

**Your Environment**

* Version used: v1.7.6
* Configuration: N/A
* Environment name and version (e.g. Kubernetes? What version?): Ubuntu 20.04
* Server type and version:  Ubuntu 20.04.2 LTS (Focal Fossa)
* Operating System and version: Ubuntu 20.04.2 LTS (Focal Fossa)
* Filters and plugins: N/A

**Additional context**

c-ares may affect this issue.

I tested some commit.
- https://github.com/fluent/fluent-bit/tree/1924d52448933044240a6bf1ee5e745c2d1d0fb4
  - OK
  -  `lib: c-ares: add new dns async library`
- https://github.com/fluent/fluent-bit/tree/bad6570a0ced3bdaf9aa56f16dc37449ad17ca37
  - Error! 
  - `network: move static function order`

Input plugin works correctly like `in_http`.

**Comment by nokute78** [comment]
@edsiper @leonardo-albertovich  can you check this?

**Comment by leonardo-albertovich** [comment]
You are correct @nokute78, it seems to be a case where c-ares does not issue a query and directly invokes the result callback without creating a socket which is why flb_net_getaddrinfo hangs after calling flb_coro_yield.

I have drafted a fix and will create a branch and pull request as well as a report so @edsiper can review it when he gets a chance.

**Comment by nokute78** [comment]
@leonardo-albertovich Thank you for quick response!

**Comment by leonardo-albertovich** [comment]
No worries, pull request sent!

**Comment by nokute78** [comment]
Thank you.
Note: #3557

**Comment by agup006** [comment]
Did this ship in 1.7.7, can we close?

**Comment by joaomnmoreira** [comment]
I'm trying to use out_forward in Fluent Bit v1.7.6 and i confirm this issue also... For several days i'm stuck in this! =/

**Comment by edsiper** [comment]
Fixed in v1.7.7 / https://www.fluentbit.io/announcements/v1.7.7/

**Comment by nokute78** [comment]
@leonardo-albertovich  @edsiper  @agup006 This issue fixed by #3557 and v1.7.7 is fixed version.
Thank you.

But I found memory leak of c-ares and opened #3572.
